mne/io/edf/edf.py

```diff
@@ -114,6 +114,11 @@ def _read_segment_file(self, data, idx, fi, start, stop, cals, mult):
         annotmap = self._raw_extras[fi]['annotmap']
         subtype = self._raw_extras[fi]['subtype']
 
+        # Interpolating stim chan except it later gets overwritten.
+        if ((not (annot and annotmap or tal_channel is not None) and
+                (stim_channel in sel))):
+            warn('Interpolating stim channel. Events may jitter.')
```

Comment:
> sel just determines what channels are being read at the moment.

Yes, but I guess we don't want a warning if the stim chan isn't picked?

Feel free to take over ;)